lib/rest-adapter.js

```diff
@@ -359,7 +359,9 @@ RestAdapter.errorHandler = function(options) {
         err.status = err.statusCode = 500;
       }
 
-      res.statusCode = err.statusCode || err.status || 500;
+      if (res.statusCode === undefined || res.statusCode === 200) {
```

Comment:
How about other 2xx status codes, like 204 No Content? I am proposing to rewrite the condition to `res.statusCode === undefined || res.statusCode < 400`